opm/simulators/wells/WellState.hpp

```diff
@@ -373,6 +383,32 @@ namespace Opm
             }
 
             const int num_perf_this_well = well_info.communication().sum(well_perf_data_[w].size());
+            if ( well.isProducer() ) { 
+                int sz = perforationRates.size();
+                int connpos = 0;
+                for (int i = 0; i < w; ++i){
+                    connpos += well_perf_data_[i].size();
+                }
+                connpos *= np;
+                double weighted_temperature = 0.0;              
+                double total_weight = 0.0;  
+                if (sz > 0){
+                   for (int i = 0; i < num_perf_this_well; ++i){
+                        int perf_cell_index = well_perf_data_[w][i].cell_index;
+                        double weight_factor = 0.0;
+                        for (int p = 0; p < np; ++p) {     
+                            weight_factor += cellDensity[perf_cell_index][p]  * perforationRates[connpos + i*np + p]/cellBinv[perf_cell_index][p] * cellInternalEnergy[perf_cell_index][p]/cellTemperatures[perf_cell_index];
+                        }
+                        total_weight += weight_factor;
+                        weighted_temperature += weight_factor * cellTemperatures[perf_cell_index];
+                    }
+                    temperature_[w] = weighted_temperature/total_weight;
```

Comment:
We need a 
```
weighted_temperature = well_info.communication().sum(weighted_temperature)
total_weight = well_info.communication().sum(total_weight);
```
before the division to account for wells which have perforation on different processes.

Reply:
Thanks for pointing out this. I have included your suggestion.

Reply:
Thanks a lot.